components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -314,7 +315,15 @@ public long getTimestamp(String date) {
     public List<KeyManagerConfigurationDTO> getKeyManagerConfigurationsByOrganization(String organization)
             throws APIManagementException {
 
-        KeyMgtRegistrationService.registerDefaultKeyManager(organization);
+        // For Choreo scenario (Choreo organization uses the same super tenant Resident Key Manager
+        // Hence no need to register the default key manager per organization)
+        String givenOrganization = organization;
+        if (MultitenantConstants.SUPER_TENANT_ID != APIUtil.getInternalOrganizationId(givenOrganization)) {
```

Comment:
let's add the change we discussed to register km for super tenant

Reply:
Sure. Fixed via 419bad1d1bac6487be3fcdfdd827331c583dde5f